lib/sshkit/scp/upload.ex

```diff
@@ -93,15 +93,22 @@ defmodule SSHKit.SCP.Upload do
   defp connection_handler(options) do
     fn message, state ->
       case message do
-        {:data, _, 0, <<@warning, data :: binary>>} -> warning(options, state, data)
-        {:data, _, 0, <<@fatal, data :: binary>>} -> fatal(options, state, data)
+        {:data, _, 0, <<@warning, data::binary>>} ->
+          warning(options, state, data)
+        {:data, _, 0, <<@fatal, data::binary>>} ->
+          fatal(options, state, data)
         {:data, _, 0, <<@normal>>} ->
           handle_data(state, options)
         {:data, _, 0, data} ->
           handle_error_data(state, options, data)
-        {:exit_status, _, status} -> exited(options, state, status)
-        {:eof, _} -> eof(options, state)
-        {:closed, _} -> closed(options, state)
+        {:data, _, 1, data} ->
```

Comment:
Note to our future selves: the integer value is the [data type](https://erlang.org/doc/man/ssh_connection.html#data-types) of the message:
> The valid values are 0 ("normal") and 1 ("stderr")

Up until now, we did not handle messages on `stderr`.